src/errors/index.js

```diff
@@ -0,0 +1,85 @@
+import * as CODES from './codes';
+import * as TEMPLATES from '../templates';
+
+
+function getErrorCode (errorClass) {
+    const code = Object
+        .keys(CODES)
+        .find(currentCode => CLASSES[currentCode] === errorClass); // eslint-disable-line no-use-before-define
+
+    return code || CODES.E000;
+}
+
+function getErrorInfo (errorClass) {
+    const code = getErrorCode(errorClass);
+
+    return {
+        constructor: CLASSES[code],  // eslint-disable-line no-use-before-define
+        message:     MESSAGES[code], // eslint-disable-line no-use-before-define
+
+        code
+    };
+}
+
+class BasicError extends Error {
+    constructor (data = {}) {
+        const { code, constructor, message }  = getErrorInfo(new.target);
+
+        super(message(data));
+
+        this.name = constructor.name;
+        this.data = data;
+        this.code = code;
+    }
+}
+
+export class BrowserPathNotSetError extends BasicError {
+    constructor () {
```

Comment:
(minor) Can we omit this empty constructor?

Reply:
yep